driver/SevenSegment.h

```diff
@@ -0,0 +1,45 @@
+#ifndef SEVENSEGMENTHEADER_H  
+#define SEVENSEGMENTHEADER_H  
+
+#include "stm32xx_hal.h"
+
+#define NUMBER_SEVEN_SEGMENTS 2
+#define NUMBER_SEGMENTS 7
```

Comment:
You have this `#define` in the `.c` and `.h`. To limit confusion it's prob best to have them in one of the files. I think it makes sense to keep them here if as a forward declaration when you define the led array.